File: pset2/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

void count_letters_words_and_sentences(string text);
void calculate_grade(float letters, float words, float sentences);
void print_grade(int index);

int main(void) {
    string text = get_string("Text: ");
    count_letters_words_and_sentences(text);
}

void count_letters_words_and_sentences(string text) {
    float letters = 0;
    float spaces = 0;
    float sentences = 0;
    for (int i = 0; i < strlen(text); i++) {
        if (text[i] == ' ') {
            spaces++;
        }
        if (text[i] == '.' || text[i] == '?' || text[i] == '!') {
            sentences++;
        }
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z')) {
            letters++;
        }
    }
    int words = spaces + 1;
    calculate_grade(letters, words, sentences);
}

void calculate_grade(float letters, float words, float sentences) {
    float lettersper100words = (letters / words) * 100;
    float sentencesper100words = (sentences / words) * 100;
    int index = round(0.0588 * lettersper100words - 0.296 * sentencesper100words - 15.8);
    print_grade(index);
}

void print_grade(int index) {
    if (index < 1) {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 15) {
        printf("Grade %i\n", index);
    }
    else {
        printf("Grade 16+\n");
    }
}

File: pset6/readability.py
from cs50 import get_string

text = get_string('Text: ')

letters = 0
spaces = 0
sentences = 0

for character in text:
    if (character >= 'a' and character <= 'z') or (character >= 'A' and character <= 'Z'):
        letters += 1
    if character == ' ':
        spaces += 1
    if character == '.' or character == '!' or character == '?':
        sentences += 1

words = spaces + 1
letters_per_100_words = (letters / words) * 100
sentences_per_100_words = (sentences / words) * 100
cl_index = round(0.0588 * letters_per_100_words - 0.296 * sentences_per_100_words - 15.8)

if cl_index < 1:
    print('Before Grade 1')
elif cl_index < 16:
    print('Grade ' + str(cl_index))
else:
    print('Grade 16+')